postprocessors/sandbox/widebeam_downsample2normalscan.py

```diff
@@ -0,0 +1,217 @@
+        -------
+        record: OrderedDict
+            hdf5 record, downsampled to one beam
+        """
+        beam2keep = 0
+        first_min = dt.datetime.fromtimestamp(record['sqn_timestamps'][0]).replace(second =0, microsecond = 0)
+        timestamp = dt.datetime.fromtimestamp(record['sqn_timestamps'][0])
+        diff = abs(first_min - timestamp).total_seconds()/record['int_time']
+        beam2keep = int(round(diff))
+        if beam2keep >= 16:
+            beam2keep = 0
+
+        #Seperate the beam to keep
+        record['beam_nums'] = np.array([np.uint32(beam2keep)])
+        record['beam_azms'] = np.array([record['beam_azms'][beam2keep]])
+        try:
+            record['data_dimensions'][0] = 1
+        except:
+            record['correlation_dimensions'][0] = 1
+        record['main_acfs'] = record['main_acfs'][beam2keep, :, :].reshape(1, record['data_dimensions'][1], record['data_dimensions'][2])
+        record['intf_acfs'] = record['intf_acfs'][beam2keep, :, :].reshape(1, record['data_dimensions'][1], record['data_dimensions'][2])
+        record['xcfs'] = record['xcfs'][beam2keep, :, :].reshape(1, record['data_dimensions'][1], record['data_dimensions'][2])
+        if beam2keep == 0:
+            record['scan_start_marker'] = True
+        else:
+            record['scan_start_marker'] = False
+        return record
+
+    @staticmethod
+    def dmap_to_dmap(file_to_process: str, processed_file: str, **kwargs) -> dict:
+        """
+        Checks what beam index is associated to timestamp and keeps only that beam from a set of 16 records.
+
+        Parameters
+        ----------
+        all_records: list
+            list of timestamps
+        record: dict
+            dictionary containing rawacf data, where the key is timestamp and each entry is a list of 16 beams
+            corresponding to one integration time/full FOV
+
+        Returns
+        -------
+        beamedrec: list
+            The downsampled record
+        """
+
+        sdarn_read = pydarnio.SDarnRead(file_to_process)
+        data = sdarn_read.read_rawacf()
+
+        record = dict()
+        all_records = []  # record names
+
+        for rec in data:  # Find the record names
+            all_records.append(
+                str(rec['time.yr']) + str(rec['time.mo']) + str(rec['time.dy']) + str(rec['time.hr']) + str(
+                    rec['time.mt']) + str(rec['time.sc']) + str(rec['time.us']))
+
+        all_records = list(dict.fromkeys(all_records))
+
+        for i in all_records:  # reformat the records in 16 records per entry to better visualise FullFOV
+            beam_rec = []
+            for rec in data:
+                rec_time = str(rec['time.yr']) + str(rec['time.mo']) + str(rec['time.dy']) + str(
+                    rec['time.hr']) + str(rec['time.mt']) + str(rec['time.sc']) + str(rec['time.us'])
+                if rec_time == i:
+                    beam_rec.append(rec)
+            record[i] = beam_rec
```

Comment:
```suggestion
        for tstamp in timestamps:  # group all records with identical timestamps
            concurrent_recs = []
            for rec in data:
                rec_time = get_timestamp(rec)
                if rec_time == tstamp:
                    concurrent_recs.append(rec)
            grouped_records.append(concurrent_recs)
```
I wouldn't recommend using `i` as a variable name unless it's an index; in this case, it was a timestamp, so it should have a more descriptive name